tests/upgrades/test_activation_key.py

```diff
@@ -0,0 +1,143 @@
+"""Test Activation Key related Upgrade Scenario's
+
+:Requirement: Upgraded Satellite
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: API
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+
+from nailgun import entities
+from requests.exceptions import HTTPError
+from robottelo.test import APITestCase
+from upgrade_tests import pre_upgrade, post_upgrade
+
+
+class scenario_positive_activation_key(APITestCase):
+    """Activation key is intact post upgrade and verify activation key update/delete
+
+    :steps:
+
+        1. In Preupgrade Satellite, create activation key with different entities
+        2. Upgrade the satellite to next/latest version
+        3. Update existing activation key with different entities
+        4. Postupgrade, Verify the activation key is intact, update and delete
+
+    :expectedresults: Activation key should create, update and delete successfully.
+    """
+    @classmethod
+    def setUpClass(cls):
+        cls.ak_name = 'preupgrade_activation_key'
+        cls.org_name = 'preupgrade_ak_org_name'
+        cls.cv_name = 'preupgrade_ak_cv'
+
+    @pre_upgrade
+    def test_pre_create_activation_key(self):
+        """Activation key with different entities are created
+
+        :id: preupgrade-a7443b54-eb2e-497b-8a50-92abeae01496
+
+        :steps:
+
+            1. In Preupgrade Satellite, Create activation key with different entities.
```

Comment:
If Its 1 step, then dirrectly append it with steps

Reply:
> If Its 1 step, then dirrectly append it with steps

@jyejare sorry, but I couldn't get you what change is required?